packages/babel-parser/src/parser/error-message.js

```diff
@@ -134,18 +134,22 @@ export const ErrorMessages = makeErrorTemplates(
     ParamDupe: "Argument name clash.",
     PatternHasAccessor: "Object pattern can't contain getter or setter.",
     PatternHasMethod: "Object pattern can't contain methods.",
-    PipelineBodyNoArrow:
+    PipeBodyIsTight:
+      "Unexpected %0 after pipeline body; any %0 expression acting as Hack-style pipe body must be parenthesized due to its loose operator precedence.",
+    // The `PipeSmartMix` error messages are retained for backwards compatibility
+    // with the deprecated smart-mix pipe operator proposal plugin.
+    PipeSmartMixBodyCannotBeArrow:
```

Comment:
```suggestion
    PipelineBodyNoArrow:
```

We should keep the old error message codes now that they are public. After this PR is merged, you could open a new PR to put all the "smart mix" stuff behind a `if (!process.env.BABEL_8_BREAKING)` flag so that they won't be included when working on Babel 8.

Reply:
This has been addressed in 86c18d3dba0972d2447eff2bd23fe3422d95efcb.